src/main/java/nl/tudelft/contextproject/saveLoad/LoadScript.java

```diff
@@ -0,0 +1,366 @@
+     * Assumes that the start element of this section has already been read
+     * and is inserted into this method as the presetStart parameter.
+     * @param presetStart The start element of the 'preset' section.
+     * @return the loaded Preset object.
+     * @throws XMLStreamException when an error occurs in the XML.
+     */
+    private static Preset loadPreset(StartElement presetStart) throws XMLStreamException {
+        int id = -1;
+        String description = "";
+        String imgLoc = "";
+        AtomicReference<CameraSettings> toSet = new AtomicReference<CameraSettings>();
+        while (reader.hasNext()) {
+            XMLEvent event = reader.nextEvent();
+            if (event.isStartElement()) {
+                StartElement start = event.asStartElement();
+                if ("id".equals(start.getName().getLocalPart())) {
+                    XMLEvent idEvent = reader.nextEvent();
+                    if (idEvent.isCharacters()) {
+                        id = Integer.parseInt(idEvent.asCharacters().getData());
+                    } else {
+                        throw new XMLStreamException("Preset ID not preset.");
+                    }
+                } else if ("description".equals(start.getName().getLocalPart())) {
+                    XMLEvent descEvent = reader.nextEvent();
+                    if (descEvent.isCharacters()) {
+                        description = descEvent.asCharacters().getData();
+                    }
+                } else if ("imgLoc".equals(start.getName().getLocalPart())) {
+                    XMLEvent imgLocEvent = reader.nextEvent();
+                    if (imgLocEvent.isCharacters()) {
+                        imgLoc = imgLocEvent.asCharacters().getData();
+                    }
+                } else if ("cameraSettings".equals(start.getName().getLocalPart())) {
+                    toSet.set(loadCameraSettings(start));
+                }
+            }
+            if (event.isEndElement()) {
+                EndElement end = event.asEndElement();
+                if ("preset".equals(end.getName().getLocalPart())) {
+                    break;
+                }
+            }
+        }
+        if (toSet.get() != null) {
+            Preset preset;
+            try {
+                Class<?> c = Class.forName(presetStart.getAttributeByName(new QName("type")).getValue());
+                Constructor<?> constructor = c.getConstructor(CameraSettings.class, int.class);
+                preset = (Preset) constructor.newInstance(toSet.get(), id);
+            } catch (Exception e) {
+                e.printStackTrace();
+                throw new XMLStreamException("Instantiating preset failed.", e);
+            }
+            preset.setDescription(description);
+            preset.setImageLocation(imgLoc);
+            return preset;
+        } else {
+            throw new XMLStreamException("No camera settings found in preset");
+        }
```

Comment:
You can probably put this if statement in another method. The current method is already too long.